silkworm/db/datastore/snapshots/btree/btree_index.cpp

```diff
@@ -76,8 +76,8 @@ std::optional<BTreeIndex::Cursor> BTreeIndex::seek(ByteView seek_key, const KVSe
     if (key.compare(seek_key) >= 0) {
         return BTreeIndex::Cursor{
             this,
-            std::move(key),
-            std::move(value),
+            key,
+            value,
```

Comment:
the move is intentional here

please update the Cursor constructor to take Bytes instead (and move into Cursor::key_/Cursor::value_)